tools/indexers/helpers.js

```diff
@@ -39,6 +40,11 @@ function indexModel(model, index, indexType, _queryParams, fetchOptions, entitie
         .then((entities) => _bulkIndexEntities(entities, index, indexType))
         // eslint-disable-next-line no-loop-func
         .then((resp) => {
+          if (resp && resp.errors) {
+            const failedModels = _.filter(resp.items, 'index.error');
+            console.error(`${failedModels.length} failed to reindex.`);
+            throw _.map(failedModels, (item) => item.index);
+          }
```

Comment:
`client.bulk()` returns the following `Object` on error:

```json
{
  "took": 12,
  "errors": true,
  "items": [
    {
      "index": {
        "_index": "trials_2017-04-05_1491385001035",
        "_type": "trial",
        "_id": "05cc77ad-5575-4c04-9309-4c64d5144b07",
        "status": 400,
        "error": {
          "type": "strict_dynamic_mapping_exception",
          "reason": "mapping set to strict, dynamic introduction of [identifiers] within [trial] is not allowed"
        }
      }
    },
    {
      "index": {
        "_index": "trials_2017-04-05_1491385001035",
        "_type": "trial",
        "_id": "475456f3-23bc-4f5e-9d19-51f4a1165540",
        "status": 400,
        "error": {
          "type": "strict_dynamic_mapping_exception",
          "reason": "mapping set to strict, dynamic introduction of [identifiers] within [trial] is not allowed"
        }
      }
    }
  ]
}
```

The docs for the latest version [include an example response of the Bulk API](https://www.elastic.co/guide/en/elasticsearch/reference/5.x/docs-bulk.html) which looks the same as :point_up:. The [docs for our version](https://www.elastic.co/guide/en/elasticsearch/reference/2.3/docs-bulk.html) don't include example response but they do mention: 

> The response to a bulk action is a large JSON structure with the individual results of each action that was performed. The failure of a single action does not affect the remaining actions.